src/Runner/Parallel/ParallelConfig.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use Fidry\CpuCoreCounter\CpuCoreCounter;
+use Fidry\CpuCoreCounter\Finder\DummyCpuCoreFinder;
+use Fidry\CpuCoreCounter\Finder\FinderRegistry;
+
+/**
+ * @author Greg Korba <greg@codito.dev>
+ */
+final class ParallelConfig
+{
+    private const DEFAULT_FILES_PER_PROCESS = 10;
+    private const DEFAULT_PROCESS_TIMEOUT = 120;
+
+    private int $filesPerProcess;
+    private int $maxProcesses;
+    private int $processTimeout;
+
+    public function __construct(
+        int $maxProcesses = 2,
+        int $filesPerProcess = self::DEFAULT_FILES_PER_PROCESS,
+        int $processTimeout = self::DEFAULT_PROCESS_TIMEOUT
+    ) {
+        if ($maxProcesses <= 0 || $filesPerProcess <= 0 || $processTimeout <= 0) {
+            throw new ParallelisationException('Invalid parallelisation configuration: only positive integers are allowed');
```

Comment:
OK to have dedicated exception for runtime logic error, but here it's misconfiguration error - I would suggest to use `InvalidArgumentException`, which we use for any invalid configuration input

Reply:
What about using `@param positive-int $maxProcesses` on the parameters and remove the checks? This assumes consumers always rely on static analysis though.

Reply:
I've changed the exception and added narrowed PHPStan types (`positive-int`). I would rather keep the explicit check and exception throwing and rely on PHPStan only for development.